File: libraries/csharp/Storage1024.cs
using System;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;

namespace Storage1024Lib
{
    public class Storage1024
    {
        private string apiBase = "https://storage1024.onrender.com/api";
        private string token;
        private string userID;
        private static readonly HttpClient client = new HttpClient();

        public void SetToken(string t) => token = t;
        public void SetUserID(string id) => userID = id;

        public async Task<string> GetGV(string name)
        {
            var request = new HttpRequestMessage(HttpMethod.Get, $"{apiBase}/projects/{userID}/gv/{name}");
            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token);
            var response = await client.SendAsync(request);
            return await response.Content.ReadAsStringAsync();
        }

        public async Task<string> AddGV(string name, string value)
        {
            var request = new HttpRequestMessage(HttpMethod.Post, $"{apiBase}/projects/{userID}/gv");
            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token);
            var json = $"{{\"alias\":\"{name}\", \"value\":\"{value}\"}}";
            request.Content = new StringContent(json, Encoding.UTF8, "application/json");
            var response = await client.SendAsync(request);
            return await response.Content.ReadAsStringAsync();
        }
    }
}

File: libraries/ruby/storage1024.rb
require 'net/http'
require 'uri'
require 'json'

class Storage1024
  def initialize
    @api_base = "https://storage1024.onrender.com/api"
    @token = nil
    @user_id = nil
  end

  def set_token(t)
    @token = t
  end

  def set_user_id(id)
    @user_id = id
  end

  def get_gv(name)
    uri = URI.parse("#{@api_base}/projects/#{@user_id}/gv/#{name}")
    request = Net::HTTP::Get.new(uri)
    request["Authorization"] = "Bearer #{@token}"

    response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
      http.request(request)
    end
    response.body
  end

  def add_gv(name, value)
    uri = URI.parse("#{@api_base}/projects/#{@user_id}/gv")
    request = Net::HTTP::Post.new(uri)
    request["Authorization"] = "Bearer #{@token}"
    request["Content-Type"] = "application/json"
    request.body = { alias: name, value: value }.to_json

    response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
      http.request(request)
    end
    response.body
  end
end
